askMenuOption.cpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include <iostream>
+#include <map>
+
+#include "askMenuOption.h"
+
+void AskMenuOption::execute() {
+	std::cout << "Choose option: ";
+}
+
+PossibleState AskMenuOption::leave() {
+	int inputOption;

```

Comment:
Leave teraz robi dwie rzeczy biznesowe.
Odpytuje użytkownika i mapuje na odpowiedni stan.
Raczej chodzi nam o to żeby wybranie i parsowanie opcji stanu zostało zrobione w execute. W leave tylko wyciągnięcie z mapy.

Niech mapa będzie statycznym memberem klasy bo teraz tworzysz ją za każdym razem. InputOption memberem obiektu i do niego wklep to co użytkownik podał. W leave tylko wyciągnięcie z mapy.